docs/source/migrating.rst

```diff
@@ -0,0 +1,189 @@
+-  Elsewhere, ``$XDG_DATA_HOME`` is respected, with the default of
+   ``~/.local/share/jupyter``
+
+In all cases, ``JUPYTER_DATA_DIR`` can be used to specify this location
+explicitly. Data files installed system-wide (e.g. in
+``/usr/local/share/jupyter``) have not changed. Per-user installation of
+these has changed from going in .ipython to the above locations:
+
+Profiles
+~~~~~~~~
+
+IPython has the concept of **profiles**, which are collections of
+configuration and runtime files. Inside the IPython directory
+(``~/.ipython``), there are directories with names like
+``profile_default`` or ``profile_demo``. In each of these are
+configuration files (``ipython_config.py``,
+``ipython_notebook_config.py``) and runtime files (``history.sqlite``,
+``security/kernel-*.json``). Profiles could be used to switch between
+configurations of IPython. **Jupyter does not have profiles**.
+
+People could use commands like ``ipython notebook --profile demo`` to
+set the profile for *both* the notebook server and the IPython kernel.
+This is no longer possible in one go with Jupyter, just like it wasn't
+possible in IPython 3 for any other kernels. If you wanted to change the
+notebook configuration, you can set the ``JUPYTER_CONFIG_DIR``:
+
+::
+
+    JUPYTER_CONFIG_DIR=./jupyter_config jupyter notebook
+
+If you just want to change the config file, you can do:
+
+::
+
+    jupyter notebook --config=/path/to/myconfig.py
+
+If, instead, you do want to change the IPython kernel's profile, you
+can't do this at the server command-line anymore. Kernel arguments must
+be done by modifying the kernelspec. The good thing about this is that
+you can do it without relaunching the server. Kernelspec changes take
+effect every time you start a new kernel. The less good thing is that
+there isn't a great way to modify the kernelspecs. You can see where the
+file is with ``jupyter kernelspec list``, and then modify
+``kernels/python3/kernel.json`` by hand.
+
+Installation
+------------
+
+I want everything
+~~~~~~~~~~~~~~~~~
+
+In IPython 3, to get everything you would type:
+
+::
+
+    pip install "ipython[all]"
+
+In Jupyter, there is a metapackage that means the same thing:
+
+::
+
+    pip install jupyter
+
+Just the notebook, please
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you wanted a subset of the functionality, say
+
+::
+
+    pip install "ipython[notebook]"
+
+you would now install the ``notebook`` package:
+
+::
+
+    pip install notebook
+
+Notebook extensions
+~~~~~~~~~~~~~~~~~~~
+
+Notebook extensions used to be installed with
+
+::
+
+    ipython install-nbextension [--user] EXTENSION
+
+They are now installed with
+
+::
+
+    jupyter nbextension install [--user] EXTENSION
+
+They will go in the ``JUPYTER_DATA_DIR`` above if a ``--user`` install
+is specified, otherwise they will go in a system-wide location (e.g.
+``/usr/local/share/jupyter/nbextensions``). Installation **SHOULD NOT**
+be done by manually guessing where the files should go.
+
```

Comment:
Ahh, got it - it the automigration triggered by running jupyter migrate, or
does that even run itself? This should be clarified.

On Mon, Sep 14, 2015 at 10:05 AM, Min RK <notifications@github.com> wrote:

> In docs/source/migrating.rst
> <https://github.com/jupyter/jupyter/pull/34#discussion_r39418755>:
>
> > +
> > +::
> > +
> > +    ipython install-nbextension [--user] EXTENSION
> > +
> > +They are now installed with
> > +
> > +::
> > +
> > +    jupyter nbextension install [--user] EXTENSION
> > +
> > +They will go in the ``JUPYTER_DATA_DIR`` above if a ``--user`` install
> > +is specified, otherwise they will go in a system-wide location (e.g.
> > +``/usr/local/share/jupyter/nbextensions``). Installation **SHOULD NOT**
> > +be done by manually guessing where the files should go.
> > +
>
> I'll make it clear that everything covered here will be auto-migrated.
> Under normal circumstances, no users will have to take any actions at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter/pull/34/files#r39418755>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com